s3file/forms.py

```diff
@@ -19,65 +20,93 @@ class S3FileInput(ClearableFileInput):
     """FileInput that uses JavaScript to directly upload to Amazon S3."""
 
     needs_multipart_form = False
-    signing_url = reverse_lazy('s3file-sign')
-    template = (
-        '<div class="s3file" data-policy-url="{policy_url}">'
-        '{input}'
-        '<input name="{name}" type="hidden" />'
-        '<div class="progress progress-striped active">'
-        '<div class="progress-bar" />'
-        '</div>'
-        '</div>'
-    )
-
-    def render(self, name, value, attrs=None):
-        parent_input = super(S3FileInput, self).render(name, value, attrs=None)
-        parent_input = parent_input.replace('name="{}"'.format(name), '')
-        output = self.template.format(
-            policy_url=self.signing_url,
-            input=parent_input,
-            name=name,
-        )
-        return mark_safe(output)
-
-    def is_initial(self, value):
-        return super(S3FileInput, self).is_initial(value)
-
-    def value_from_datadict(self, data, files, name):
-        filename = data.get(name, None)
-        if not self.is_required and CheckboxInput().value_from_datadict(
-                data, files, self.clear_checkbox_name(name)):
-            if filename:
-                # If the user contradicts themselves (uploads a new file AND
-                # checks the "clear" checkbox), we return a unique marker
-                # object that FileField will turn into a ValidationError.
-                return FILE_INPUT_CONTRADICTION
-            # False signals to clear any existing value, as opposed to just None
-            return False
-        if not filename:
-            return None
+    mime_type = None
+
+    def __init__(self, attrs=None):
+        self.expires = timedelta(seconds=settings.SESSION_COOKIE_AGE)
+        self.access_key = settings.AWS_ACCESS_KEY_ID
+        self.secret_access_key = settings.AWS_SECRET_ACCESS_KEY
+        self.bucket_name = settings.AWS_STORAGE_BUCKET_NAME
+        self.upload_path = getattr(settings, 'S3FILE_UPLOAD_PATH', os.path.join('tmp', 's3fine'))
+        super(S3FileInput, self).__init__(attrs=attrs)
         try:
-            upload = default_storage.open(filename)
-        except IOError:
-            logger.exception('File "%s" could not be found.', filename)
-            return False
+            self.mime_type = self.attrs['accept']
+        except KeyError:
+            pass
+
+    def get_expiration_date(self):
+        expiration_date = datetime.utcnow() + self.expires
+        return expiration_date.strftime('%Y-%m-%dT%H:%M:%S.000Z')
+
+    def build_attrs(self, *args, **kwargs):
+        attrs = super(S3FileInput, self).build_attrs(*args, **kwargs)
+        defaults = {
+            'data-policy': force_text(self.get_policy()),
+            'data-signature': self.get_signature(),
+            'data-key': self.upload_folder,
+            'data-s3-url': 'https://s3.amazonaws.com/%s' % self.bucket_name,
```

Comment:
`https://s3.amazonaws.com/` works for `us-east-1` and maybe some other regions but didn't work for me for `eu-west-1`

Reply:
dang. can you show me tomorrow?

Reply:
Here's the list of regions and s3 endpoints: http://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region 

I think you would need an extra setting where to specify S3 region for the bucket if you want to support regions other than `us-east-1`